flink-ml-core/src/test/java/org/apache/flink/ml/api/PipelineTest.java

```diff
@@ -95,4 +109,59 @@ public void testPipeline() throws Exception {
         // Executes the loaded Pipeline and verifies that it produces the expected output.
         TestUtils.executeAndCheckOutput(env, loadedEstimator, inputs, output, null, null);
     }
+
+    @Test
+    public void testSupportServable() {
+        SumEstimator estimatorA = new SumEstimator();
+        UnionAlgoOperator algoOperatorA = new UnionAlgoOperator();
+        SumModel modelA = new SumModel();
+        SumModel modelB = new SumModel();
+
+        List<Stage<?>> stages = Arrays.asList(modelA, modelB);
+        PipelineModel pipelineModel = new PipelineModel(stages);
+        assertTrue(pipelineModel.supportServable());
+
+        stages = Arrays.asList(estimatorA, modelA);
+        pipelineModel = new PipelineModel(stages);
+        assertFalse(pipelineModel.supportServable());
+
+        stages = Arrays.asList(algoOperatorA, modelA);
+        pipelineModel = new PipelineModel(stages);
+        assertFalse(pipelineModel.supportServable());
+    }
+
+    @Test
+    public void testLoadServable() throws Exception {
+        SumModel modelA = new SumModel().setModelData(tEnv.fromValues(10));
+        SumModel modelB = new SumModel().setModelData(tEnv.fromValues(20));
+        SumModel modelC = new SumModel().setModelData(tEnv.fromValues(30));
+
+        List<Stage<?>> stages = Arrays.asList(modelA, modelB, modelC);
+        Model<?> model = new PipelineModel(stages);
+
+        PipelineModelServable servable =
+                saveAndLoadServable(tEnv, model, tempFolder.newFolder().getAbsolutePath());
```

Comment:
Compared with the disadvantage that users cannot tell the test method directly, I'd rather keep the code clean by wrapping the common logic in a util method, like `TestUtils#saveAndReload`. I believe a developer should notice that the `loadServable` is tested in `testLoadServable` by method name.

Reply:
I am not sure developers can "easily" notice that this `PipelineModel#loadServable` is tested in `PipelineTest#testLoadServable`. IMO it is much easier for developers to know where this method is tested via IDE if this method is directly invoked here.

Maybe we can let a third person break the tie. @zhipeng93 can you comment on which solution you prefer?

Reply:
I aggree with @lindong28 that using `PipelineModel#loadServable()` is better for readability. 

However, I also noticed that we are using refelctions to test `saveAndReload` for all other Model/Transformers (e.g., KmeansTest [1]). In this case, using reflections for testing `PipelineModel#loadServable()` here is ok to me. If we want better readability, we can update all the existing tests in a later PR.

@jiangxin369 @lindong28 What do you think?


[1] https://github.com/apache/flink-ml/blob/master/flink-ml-lib/src/test/java/org/apache/flink/ml/clustering/KMeansTest.java#L217

Reply:
I agree we can also update the existing `TestUtils#saveAndReload` to make the code more readable.

This can be achieved by passing e.g. `KMeans#load` as parameter of type `BiFunctionWithException<StreamTableEnvironment, String, T, IOException>` to `TestUtils#saveAndReload`.

Reply:
I agree with the solution and I've updated the PR.